### PR TITLE
Issue #1590: Gradle build fails when run in parallel - add dependency…

### PR DESCRIPTION
**Pull request by RyanHolstien**
… on avro schema generation to metadata models jar generation

### CODE REVIEW

**Comment by RyanHolstien** [comment]
> Even after adding this change `./gradlew clean build --parallel` fails in the same way in my local? What do I miss?

Hmm, this fixed it for me and has been consistently working for us in development. Tested it out on the open source branch a few times. We have made some other changes to our fork though which may be impacting us consistently not seeing it while we develop. You get the same error that I posted on the issue I opened?

I'll try it out on one of my other machines and see if I can still reproduce.

**Comment by keremsahin1** [comment]
```
Gradle suite > Gradle test > com.linkedin.metadata.EventUtilsTests.testAvroToPegasusMAE FAILED
    java.lang.ExceptionInInitializerError at EventUtilsTests.java:30
        Caused by: java.lang.IllegalArgumentException at EventUtilsTests.java:30

Gradle suite > Gradle test > com.linkedin.metadata.EventUtilsTests.testAvroToPegasusMCE FAILED
    java.lang.NoClassDefFoundError at EventUtilsTests.java:42

Gradle suite > Gradle test > com.linkedin.metadata.EventUtilsTests.testPegasusToAvroFailedMCE FAILED
    java.lang.NoClassDefFoundError at EventUtilsTests.java:73

Gradle suite > Gradle test > com.linkedin.metadata.EventUtilsTests.testPegasusToAvroMAE FAILED
    java.lang.NoClassDefFoundError at EventUtilsTests.java:53

Gradle suite > Gradle test > com.linkedin.metadata.EventUtilsTests.testPegasusToAvroMCE FAILED
    java.lang.NoClassDefFoundError at EventUtilsTests.java:63

5 tests completed, 5 failed
```
I'm getting above error and it's the same error as you shared.  Could it be because of gradle version? I use:

```
./gradlew --version

------------------------------------------------------------
Gradle 5.6.4
------------------------------------------------------------

Build time:   2019-11-01 20:42:00 UTC
Revision:     dd870424f9bd8e195d614dc14bb140f43c22da98

Kotlin:       1.3.41
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.14 compiled on March 12 2019
JVM:          1.8.0_212 (Oracle Corporation 25.212-b10)
OS:           Mac OS X 10.15.3 x86_64
```

**Comment by RyanHolstien** [comment]
I'm using the same gradle version. I was able to reproduce on a different machine. I had some other changes that I didn't think were necessary for the open source side on my local that I removed and I think gradle didn't clean them up properly when I ran clean build as opposed to clean -> build so it had been working. I'll push those out.

**Comment by keremsahin1** [comment]
Perfect, last diff worked! `processResources` should have a dependency on `copyOriginalMXESchemas`.

File: metadata-events/mxe-utils-avro-1.7/build.gradle [hunk 1]
```diff
@@ -27,7 +27,21 @@ task copyOriginalMXESchemas(type: Copy) {
 }
 
 compileJava.dependsOn copyOriginalMXESchemas
+processResources.dependsOn copyOriginalMXESchemas
+
+task copyOriginalMXESchemasTest(type: Copy) {
```
Comment (keremsahin1) [inline metadata-events/mxe-utils-avro-1.7/build.gradle]: Just curious, why do you need this? I guess we don't need to copy Avro schemas to test resources? I tried removing this task & its dependencies below, build still works.

Comment (RyanHolstien) [reply metadata-events/mxe-utils-avro-1.7/build.gradle]: So we needed it on our fork when we added tests to our modified consumer apps as they directly used the Avro schemas in the tests. I can remove it here if it's not needed on the OSS. I didn't think this part was needed for the test you had added which was why I removed it in the initial commit. I can remove the parts here other than the processResources dependency :)

Comment (keremsahin1) [reply metadata-events/mxe-utils-avro-1.7/build.gradle]: That sounds good. Let's remove it for now.

**Comment by mars-lan** [comment]
Thanks for the PR, @RyanHolstien. @keremsahin1 let's enable parallel build by default in `gradle.properties`?

**Pull request merged**